src/utils.ts

```diff
@@ -15,7 +15,12 @@ export function useOuterClick(
 
   React.useEffect(() => {
     function handler(event) {
-      if (dom.current?.contains(event.target)) {
+      if (
+        dom.current?.contains(event.target) ||
+        // Add support for ReactShadowRoot
+        // @ts-expect-error wrong types, the `host` property exists https://stackoverflow.com/a/25340456
+        event.target === dom.current?.getRootNode().host
```

Comment:
Events inside the shadow DOM do bubble up to `window`, however their `target` will be the host element, not the actual element, because the shadow DOM incapsulate everything inside it.